features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -218,8 +222,17 @@ bool ATHandler::get_debug() const
 
 ATHandler::~ATHandler()
 {
+    ScopedLock <ATHandler> lock(*this);
     set_file_handle(NULL);
 
+    while (_event_id > 0) {
+#if defined AT_HANDLER_MUTEX && defined MBED_CONF_RTOS_PRESENT
+        _oobCv.wait();
+#else
+        rtos::ThisThread::sleep_for(PROCESS_URC_TIME);
```

Comment:
Don't follow the logic of this `else`. If `AT_HANDLER_MUTEX` is not defined, we must be assuming single-threaded use. If we're single-threaded, sleeping won't help anything - there's no other thread to let the OOB to run to completion. As I suggested previously, this is an assert failure - we cannot destroy ourselves is there is a pending event.